### ISSUE TITLE
Delta mishandles multidimensional event shapes

### ISSUE THREAD

**Issue by aswanth-07** [pytorch/rl#4090]
## Describe the bug

`torchrl.modules.distributions.Delta` mishandles an explicitly configured multidimensional `event_shape` in two adjacent paths:

1. `log_prob()` reduces changing negative dimensions and returns the wrong batch shape.
2. `expand()` silently reclassifies one event dimension as a batch dimension in the returned distribution.

## To reproduce

```python
import torch

from torchrl.modules.distributions import Delta


param = torch.arange(24.0).reshape(2, 3, 4)
dist = Delta(param, batch_shape=(2,), event_shape=(3, 4))
expanded = dist.expand((5, 2))

print("original:", dist.batch_shape, dist.event_shape, dist.log_prob(dist.param).shape)
print("expanded:", expanded.batch_shape, expanded.event_shape, expanded.param.shape)
print("expanded log_prob:", expanded.log_prob(expanded.param).shape)
```

On current `main` this prints:

```text
original: (2,) (3, 4) torch.Size([3])
expanded: torch.Size([5, 2, 3]) torch.Size([4]) torch.Size([5, 2, 3, 4])
expanded log_prob: torch.Size([5, 2, 3])
```

## Expected behavior

The original and expanded distributions should preserve the configured event semantics:

```text
original: (2,) (3, 4) torch.Size([2])
expanded: torch.Size([5, 2]) torch.Size([3, 4]) torch.Size([5, 2, 3, 4])
expanded log_prob: torch.Size([5, 2])
```

`Delta.expand()` currently creates the returned `Delta` without forwarding `batch_shape` or `event_shape`, so the constructor falls back to inferring only the final parameter dimension as the event. Separately, `_is_equal()` reduces `-1` and then `-2` after the tensor rank has changed; reducing the current last dimension for each event axis preserves the batch dimensions.

## System information

- TorchRL: current `main` at `7294465173cb23121386db0ce115abba54fc4a28`
- PyTorch: `2.11.0+cu130` (reproduced on CPU)
- TensorDict: `0.13.0`
- OS: Windows 11

## AI assistance disclosure

I used OpenAI Codex to help inspect the implementation, construct and run the pristine reproduction, and search existing issues and pull requests for duplicates. I reviewed and reproduced the behavior before submitting this report.
